### ISSUE TITLE
[Modals]: incorrect focus order via keyboard navigation

### ISSUE THREAD

**Issue by NatalliaAlieva** [epam/UUI#2603]
## Description
Upon activation, the modal window does not receive immediate focus with keyboard navigation, as the focus moves sequentially through remaining page elements first.

## Steps to Reproduce
1. Open [Modal](https://uui.epam.com/documents?id=modals&mode=doc&category=components&theme=electric#modal_with_form) component.
2. Navigate to "Modal with Form" example with `Tab` key.
3. Activate "Modal with Form" example with `Enter` key.
4. Try using the keyboard to navigate inside the modal right after it opens.
5. Observe the behavior of the focus.

## Actual result
The focus does not immediately enter the modal window. Instead, it progresses through other elements on the underlying page before finally moving into the modal.

https://github.com/user-attachments/assets/c81e3868-604b-4fe2-a4f3-62ca1943badd

## Expected result
When the modal window is activated by pressing Enter, focus should immediately move to the first interactive element inside the modal and be visible right after the modal opens. This ensures intuitive and seamless keyboard navigation within the modal.

## Environment
* UUI version: 5.10.2
* Browser name: any
* Operating System: any

**Comment by AlekseyManetov** [comment]
Related to https://github.com/theKashey/react-focus-lock/issues/340

**Comment by AlekseyManetov** [comment]
As a temporary fix we add autofocus first active element on modal open. It's a temporary fix, because of bug in react-focus-lock lib(https://github.com/theKashey/react-focus-lock/issues/340). Ideally focus should move to the first active element only after 'Tab' key was pressed, we will come back to this solution when bug will be fixed.

**Comment by NatalliaAlieva** [comment]
> As a temporary fix we add autofocus first active element on modal open. It's a temporary fix, because of bug in react-focus-lock lib([theKashey/react-focus-lock#340](https://github.com/theKashey/react-focus-lock/issues/340)). Ideally focus should move to the first active element only after 'Tab' key was pressed, we will come back to this solution when bug will be fixed.

Released in 5.12.0 ver.

**Comment by NatalliaAlieva** [comment]
Released in 5.13.0 ver.
